server/src/middleware/loggerMiddleware.js

```diff
@@ -0,0 +1,65 @@
+import fs from 'fs';
+import morgan from 'morgan';
+import path from 'path';
+import rfs from 'rotating-file-stream';
+import chalk from 'chalk';
+
+class LoggerMiddleware {
+  constructor() {
+    const logDirectory = path.join(__dirname, '..', 'logs');
```

Comment:
This is repo root? Will cause problems with Docker container separation. API logs should be in server/logs.